x/payments/internal/types/params.go

```diff
@@ -133,18 +97,218 @@ func (p Params) String() string {
+
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("ixo factor must be not nil")
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("ixo factor must be positive: %s", v)
+	}
+	if v.GT(sdk.OneDec()) {
+		return fmt.Errorf("ixo factor too large: %s", v)
+	}
+
+	return nil
+}
+
+func validateFeeAmount(i interface{}) error {
+	v, ok := i.(sdk.Dec)
+
+	if !ok {
+		return fmt.Errorf("invalid fee amount type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("fee amount must be not nil")
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("fee amount must be positive: %s", v)
+	}
+	if v.GT(sdk.OneDec()) {
+		return fmt.Errorf("fee amount too large: %s", v)
+	}
+
+	return nil
+}
+
+func validateFeePercentage(i interface{}) error {
+	v, ok := i.(sdk.Dec)
+
+	if !ok {
+		return fmt.Errorf("invalid fee percentage type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("fee percentag must be not nil")
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("fee percentag must be positive: %s", v)
+	}
+	if v.GT(sdk.OneDec()) {
+		return fmt.Errorf("fee percentag too large: %s", v)
+	}
+
+	return nil
+}
+
+func validateClaimFeeAmount(i interface{}) error {
+	v, ok := i.(sdk.Dec)
+
+	if !ok {
+		return fmt.Errorf("invalid claim fee amount type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("claim fee amount must be not nil")
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("claim fee amount must be positive: %s", v)
+	}
+	if v.GT(sdk.OneDec()) {
+		return fmt.Errorf("claim fee amount too large: %s", v)
+	}
+
+	return nil
+}
+
+func validateEvaluationFeeAmount(i interface{}) error {
+	v, ok := i.(sdk.Dec)
+
+	if !ok {
+		return fmt.Errorf("invalid evaluation fee amount type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("evaluation fee amount must be not nil")
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("evaluation fee amount must be positive: %s", v)
+	}
+	if v.GT(sdk.OneDec()) {
+		return fmt.Errorf("evaluation fee amount too large: %s", v)
+	}
+
+	return nil
+}
+
+func validateRegistrationFeeAmount(i interface{}) error {
+	v, ok := i.(sdk.Dec)
+
+	if !ok {
+		return fmt.Errorf("invalid registration fee amount type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("registration fee amount must be not nil")
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("registration fee amount must be positive: %s", v)
+	}
+	if v.GT(sdk.OneDec()) {
+		return fmt.Errorf("registration fee amount too large: %s", v)
+	}
+
+	return nil
+}
+
+func validateAgentRegistrationFeeAmount(i interface{}) error {
+	v, ok := i.(sdk.Dec)
+
+	if !ok {
+		return fmt.Errorf("invalid agent registration fee amount type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("agent registration fee amount must be not nil")
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("agent registration fee amount must be positive: %s", v)
+	}
+	if v.GT(sdk.OneDec()) {
+		return fmt.Errorf("agent registration fee amount too large: %s", v)
```

Comment:
This check is not valid and not required.

Reply:
ok